res/controllers/Pioneer-DDJ-400.midi.xml

```diff
@@ -0,0 +1,4877 @@
+<?xml version="1.0" encoding="utf-8"?>
+<MixxxMIDIPreset schemaVersion="1" mixxxVersion="2.3">
+    <info>
+        <name>Pioneer DDJ-400</name>
+        <author>Warker/nschloe DEV</author>
+        <description>Midi Mapping for the Pioneer DDJ-400</description>
+        <wiki>https://www.mixxx.org/wiki/doku.php/pioneer_ddj-400</wiki>
+        <forums>https://www.mixxx.org/forums/viewtopic.php?f=7&amp;t=12113</forums>
+    </info>
+    <controller id="DDJ-400">
+        <scriptfiles>
+            <file filename="lodash.mixxx.js"/>
+            <file filename="midi-components-0.0.js"/>
+            <file functionprefix="PioneerDDJ400" filename="Pioneer-DDJ-400-script.js"/>
+        </scriptfiles>
+        <controls>
+            <!--
+                BROWSER Section START
+            -->
+            <control>
+                <description>BROWSE - rotate - Scroll tracklist/tree view</description>
+                <group>[Library]</group>
+                <key>MoveVertical</key>
+                <status>0xB6</status>
+                <midino>0x40</midino>
+                <options>
+                    <SelectKnob/>
+                </options>
+            </control>
+            <control>
+                <description>BROWSE +SHIFT - rotate - Zoom in/out of enlarged waveforms</description>
+                <group>[Master]</group>
+                <key>PioneerDDJ400.waveFormRotate</key>
+                <status>0xB6</status>
+                <midino>0x64</midino>
+                <options>
+                    <Script-Binding/>
+                </options>
+            </control>
+            <control>
+                <description>BROWSE - press - Move cursor between track list and tree view / open a folder</description>
+                <group>[Library]</group>
+                <key>MoveFocusForward</key> <!-- missing function to open Folder -->
+                <status>0x96</status>
+                <midino>0x41</midino>
+                <options>
+                    <Normal/>
+                </options>
+            </control>
+            <control>
+                <description>BROWSE +SHIFT - press - Move cursor between track list and tree view / close a folder
```

Comment:
This is not documented in the wiki.

Reply:
This doesn't quite work as described, either.  When SHIFT+BROWSE is pressed, cursor moves (in "reverse" order, i.e. to the left) between the track list, the tree view, the Beat Loop Size field, and the Beatjump/Loop Move Size fields for each deck.

Furthermore, if the Track List view is currently displaying an "empty view" which contains a clickable link (such as the Crate view, which has the link "Create New Create" then the cursor will move to that link also as part of the cycle.

So, in effect, SHIFT+BROWSE moves between:

Track List (or Track List Hyperlink) -> Tree View -> Beatjump/Loop Move Size (Deck 2) -> Beat Loop Size (Deck 2) -> Beatjump/Loop Move Size (Deck 1) -> Beat Loop Size (Deck 1) -> Track List (or Track List Hyperlink)

Reply:
It's documented now.